input.c

```diff
@@ -260,9 +265,9 @@ static int do_http_get(struct http_get *hg, const char *uri, int redirections)
 
 		redirloc = xstrdup(val);
 		http_get_free(hg);
-		close(hg->fd);
+		connection_close(conn);
```

Comment:
Won't http_get_free also not already close the `fd`? Causing you to close it twice. `close` won't mind, still.